src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs

```diff
@@ -109,14 +109,18 @@ private static void ReferenceNetstandardDllIfCoreClr(List<string> arguments)
 #endif
         }
 
+        private static readonly object s_createFilesLock = new object();
```

Comment:
Should this lock be inside TempDirectory so we don't get bit by this at some later point? We have plenty of other tests in the IDE layer at least that do multithreading that I now wonder if they're also broken.